test/config_test/config_test.cc

```diff
@@ -92,8 +92,17 @@ uint32_t run(const std::string& directory) {
   RELEASE_ASSERT(::chdir(directory.c_str()) == 0);
   uint32_t num_tested = 0;
   for (const std::string& filename : TestUtility::listFiles(directory, false)) {
-    ConfigTest config(filename);
+    Server::TestOptionsImpl options(filename, Network::Address::IpVersion::v6);
+    ConfigTest test1(options);
     num_tested++;
+
+    // Config flag --config-yaml is only supported for v2 configs.
+    envoy::config::bootstrap::v2::Bootstrap bootstrap;
+    if (Server::InstanceUtil::loadBootstrapConfig(bootstrap, options) ==
+        Server::InstanceUtil::BootstrapVersion::V2) {
+      ConfigTest test2(options.asConfigYaml());
+      num_tested++;
+    }
```

Comment:
Can you add a test to validate merge behavior? It can be really simple, just show that you get the union for two fields.

Reply:
Done.